[Docs] Update for image spec/fast register notes (#5726)
flyteorg/flyte: docs/user_guide/customizing_dependencies/imagespec.md

image = ImageSpec(registry="ghcr.io/flyteorg", packages=["pandas"]).force_push()
```
[flytesnacks]: https://github.com/flyteorg/flytesnacks/tree/master/examples/customizing_dependencies/

## Getting source files into ImageSpec

> Typically, getting source code files into a task's image at run time on a live Flyte backend is done through the fast registration mechanism.

Reviewer: ```suggestion
Typically, getting source code files into a task's image at run time on a live Flyte backend is done through the fast registration mechanism.

However, if your `ImageSpec` constructor specifies a `source_root` and the `copy` argument is set to something other than `CopyFileDetection.NO_COPY`, then files will be copied regardless of fast registration status.
```

Reviewer: I think it's fine to just say the same thing in two places for clarity. In the future, we can use reusable text snippets for stuff like this.

> If the `source_root` and `copy` fields to an `ImageSpec` are left blank, then whether or not your source files are copied into the built `ImageSpec` image is basically the inverse of whether fast register is used. Please see [registering workflows](https://docs.flyte.org/en/latest/flyte_fundamentals/registering_workflows.html#containerizing-your-project) for the full explanation.

Reviewer: ```suggestion
If the `source_root` and `copy` fields to an `ImageSpec` are left blank, then whether or not your source files are copied into the built `ImageSpec` image depends on whether or not you use fast registration. Please see [registering workflows](https://docs.flyte.org/en/latest/flyte_fundamentals/registering_workflows.html#containerizing-your-project) for the full explanation.
```

> Keep in mind that because files are sometimes copied into the built image, the tag that is published for an ImageSpec will change based on whether fast register is enabled, and the contents of any files copied.

Reviewer: ```suggestion
Since files are sometimes copied into the built image, the tag that is published for an ImageSpec will change based on whether fast register is enabled, and the contents of any files copied.
```
